runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -7942,44 +7938,55 @@ J9::X86::TreeEvaluator::VMnewEvaluator(
                else if (span > 3)
                   {
+                  TR::InstOpCode::Mnemonic storeOpCode;
                   if (lastSpan == 0)
                      {
-                     generateMemRegInstruction(TR::InstOpCode::MOVDMemReg, node, generateX86MemoryReference(targetReg, lastElementIndex*4 +dataOffset, cg), scratchReg, cg);
+                     storeOpCode = TR::InstOpCode::MOVDMemReg;
                      }
                   else if (lastSpan == 1)
                      {
-                     generateMemRegInstruction(TR::InstOpCode::MOVQMemReg, node,generateX86MemoryReference(targetReg, lastElementIndex*4 +dataOffset, cg), scratchReg, cg);
+                     storeOpCode = TR::InstOpCode::MOVQMemReg;
                      }
                   else
                      {
-                     generateMemRegInstruction(TR::InstOpCode::MOVDQUMemReg, node, generateX86MemoryReference(targetReg, lastElementIndex*4 +dataOffset, cg), scratchReg, cg);
+                     storeOpCode = TR::InstOpCode::MOVDQUMemReg;
                      }
+
+                  generateMemRegInstruction(storeOpCode, node, generateX86MemoryReference(targetReg, lastElementIndex*4 +dataOffset, cg), scratchReg, cg);
+
                   lastElementIndex = nextE;
                   lastSpan = 0;
                   }
                }
-            if (lastSpan == 0)
-               {
-               generateMemRegInstruction(TR::InstOpCode::MOVDMemReg, node, generateX86MemoryReference(targetReg, lastElementIndex*4 +dataOffset, cg), scratchReg, cg);
-               }
-            else if (lastSpan == 1)
-               {
-               generateMemRegInstruction(TR::InstOpCode::MOVQMemReg, node,generateX86MemoryReference(targetReg, lastElementIndex*4 +dataOffset, cg), scratchReg, cg);
-               }
-            else if (lastSpan == 2)
+
+            if (lastSpan < 3)
                {
-               TR_ASSERT(dataOffset >= 4, "dataOffset must be >= 4.");
-               generateMemRegInstruction(TR::InstOpCode::MOVDQUMemReg, node, generateX86MemoryReference(targetReg, lastElementIndex*4 +dataOffset - 4, cg), scratchReg, cg);
+               int32_t adjustedDataOffset = dataOffset;
+
+               TR::InstOpCode::Mnemonic storeOpCode;
+               if (lastSpan == 0)
+                  {
+                  storeOpCode = TR::InstOpCode::MOVDMemReg;
+                  }
+               else if (lastSpan == 1)
+                  {
+                  storeOpCode = TR::InstOpCode::MOVQMemReg;
+                  }
+               else if (lastSpan == 2)
+                  {
+                  TR_ASSERT(dataOffset >= 4, "dataOffset must be >= 4.");
+                  storeOpCode = TR::InstOpCode::MOVDQUMemReg;
+                  adjustedDataOffset -= 4;
+                  }
+
+               generateMemRegInstruction(storeOpCode, node, generateX86MemoryReference(targetReg, lastElementIndex*4 + adjustedDataOffset, cg), scratchReg, cg);
```

Comment:
I'm seeing crashes in this code where `lastSpan == -1` and `storeOpCode` is uninitialized. I think the existing code avoided generating an instruction in the `-1` case where as with this patch we avoid initializing the store but not generating the instruction. FYI @BradleyWood.

Reply:
Yes, I have found and fixed this in my latest branch (which contains a number of other changes).  I haven't pushed yet while testing is underway.